modules/media/jsx/uploadForm.js

```diff
@@ -340,9 +340,23 @@ var MediaUploadForm = React.createClass({
    */
   setFormData: function(formElement, value) {
     // Only display visits and sites available for the current pscid
+    var visitLabel=this.state.formData.visitLabel;
     if (formElement === "pscid" && value !== "") {
       this.state.Data.visits = this.state.Data.sessionData[value].visits;
       this.state.Data.sites = this.state.Data.sessionData[value].sites;
+      if (visitLabel !== null) {
```

Comment:
This checks for `null`, but visit label is undefined so the condition doesn't work.
You should do instead:
```js
if (visitLabel) {
  ...
} else {
  ...
}
```